### DISCUSSION TITLE
Can the Accordion trigger text change depending on whether the parent item is open or not?

### DISCUSSION THREAD

**Discussion by hoggworks** [radix-ui/primitives#1515] [Help] [answered]
Hey all,
I'm just looking through the Radix ui documentation, which looks pretty nice. I'm looking initially at trying out the Accordion example, and the sample on the main site is impressively robust, but I'm wondering if there's a way that the AccordionTrigger element can know whether the trigger it's in is open. My usecase is as follows:

When the item is closed, I want to be able to provide a summary of the content:
<AccordionTrigger>Contact Options - email, sms, app</AccordionTrigger>

Then, when it's open, it would just look like:
<AccordionTrigger>Contact Options</AccordionTrigger>

It looks like I can use a state variable to control which item is open, but is there an onClick I can use, or something else to broadcast when the <AccordionTrigger/> element is clicked, so I can track it externally? 

Thanks!

**Comment by benoitgrelard** [comment, marked as answer]
Hi @hoggworks,

You should be able to achieve what you are after by using the controlled API of `Accordion` (`value`, `onValueChange`).
You then are the one providing the state, and can switch your text based on whether the current `value` is the one in each item (or contains, if you allow multiple opened at the same time).

**Reply by andy-hook** [reply]
Quick example for `type="single"` that I had in my pocket https://codesandbox.io/s/accordion-change-trigger-text-when-open-q0vc25?file=/App.js

**Comment by hoggworks** [comment]
Thanks very much!
